api/src/controllers/users.ts

```diff
@@ -371,12 +381,20 @@ router.post(
 			throw new InvalidPayloadError({ reason: `"otp" is required` });
 		}
 
+		const usersService = new UsersService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const currentUser = await usersService.readOne(req.accountability.user, { fields: ['provider'] });
+		const requiresPassword = currentUser?.['provider'] === 'default';
```

Comment:
In this case we are not actively using `requiresPassword` here so I think we should move this entirely into the `service.enableTFA` function as there we are already fetching the provider info directly from the db
```suggestion
```